src/mapgen.cpp

```diff
@@ -8435,6 +8443,59 @@ bool run_mapgen_update_func( const std::string &update_mapgen_id, const tripoint
     return update_function->second[0]->update_map( omt_pos, 0, 0, miss, cancel_on_collision );
 }
 
+std::pair<std::map<ter_id, int>, std::map<furn_id, int>> get_changed_ids_from_update(
+            const std::string &update_mapgen_id )
+{
+    std::map<ter_id, int> terrains;
+    std::map<furn_id, int> furnitures;
+
+    const auto update_function = update_mapgen.find( update_mapgen_id );
+
+    if( update_function == update_mapgen.end() || update_function->second.empty() ) {
+        return std::make_pair( terrains, furnitures );
+    }
+
+    const tripoint omt_pos = tripoint( 0, 0, 0 );
+
+    tinymap fake_map;
+    fake_map.load( 0, 0, 0, false );
+    for( const tripoint &pos : fake_map.points_in_rectangle( omt_pos,
+            tripoint( MAPSIZE * SEEX, MAPSIZE * SEEY, 0 ) ) ) {
+        fake_map.furn_set( pos, f_null );
+        fake_map.ter_set( pos, t_dirt );
+        fake_map.trap_set( pos, tr_null );
+    }
+
+    const regional_settings &rsettings = overmap_buffer.get_settings( omt_pos.x, omt_pos.y,
+                                         omt_pos.z );
+    const std::string map_id = overmap_buffer.ter( omt_pos ).id().c_str();
```

Comment:
Maybe I'm blind, but it doesn't look like this gets used?

Reply:
What, me leave some scaffolding code behind when I'm figuring something out?